src/validations/request.validation.js

```diff
@@ -82,3 +86,62 @@ export const validateChangingRequestStatus = async (req, res, next) => {
   }
   next();
 };
+/**
+ * @param {req} req
+ * @param {res} res
+ * @param {next} next
+ * @returns {returns} return values
+*/
+export async function editTripRequestValidation(req, res, next) {
+  const schema = Joi.object({
+    originId: Joi.number(),
+    destinationId: Joi.number(),
+    departureDate: Joi.date().greater('now').utc().format('YYYY-MM-DD')
+      .messages({
+        'date.greater': 'Departure date should be greater than today\'s date',
+        'date.format': 'Departure date must be in YYYY-MM-DD format'
+      }),
+    returnDate: Joi.date().greater('now').utc().format('YYYY-MM-DD')
+      .messages({
+        'date.greater': 'Return date should be greater than today\'s date',
+        'date.format': 'Return date must be in YYYY-MM-DD format'
+      }),
+    travelReasons: Joi.string().min(5).trim()
+      .messages({
+        'string.empty': 'Travel reasons is not allowed to be empty',
+        'string.min': 'Travel reasons must be at least 5 characters long'
+      }),
+    accommodationId: Joi.number()
+  }).options({ abortEarly: false });
+
+
+  const { error } = schema.validate(req.body);
+  if (error) {
+    const errorMessages = error.details.map((err) => (err.message.replace(/[^a-zA-Z0-9 .-]/g, '')));
+    ResponseService.setError(400, errorMessages);
+    ResponseService.send(res);
+  }
+
+  if (req.body.originId && req.body.destinationId) {
+    const locations = [req.body.originId, req.body.destinationId];
+    await Promise.all(locations.map(async loc => {
```

Comment:
https://andela.slack.com/archives/GQ9DMDBJR/p1579014769014900?thread_ts=1579014551.014400&cid=GQ9DMDBJR

Reply:
@adesege i am not getting what you wanna mean here? Can you please clarify.

Reply:
I already gave feedback on Slack on how to efficiently use select with an array of ids. Follow the link to see the details of the feedback.

Reply:
ok